source.go

```diff
@@ -82,15 +85,43 @@ func (s *Source) Config() sdk.SourceConfig {
 }
 
 func (s *Source) Open(ctx context.Context, pos opencdc.Position) error {
-	sdk.Logger(ctx).Info().Msg("Opening DynamoDB Source...")
-
-	cfg, err := config.LoadDefaultConfig(ctx,
-		config.WithRegion(s.config.AWSRegion),
-		config.WithCredentialsProvider(aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(s.config.AWSAccessKeyID, s.config.AWSSecretAccessKey, s.config.AWSSessionToken))),
-	)
-	if err != nil {
-		return fmt.Errorf("could not load AWS config: %w", err)
-	}
+    sdk.Logger(ctx).Info().Msg("Opening DynamoDB Source...")
+
+    // Build up LoadDefaultConfig options
+    opts := []func(*config.LoadOptions) error{
+        config.WithRegion(s.config.AWSRegion),
+    }
+
+    // If the user supplied static creds, register them first
+    if s.config.AWSAccessKeyID != "" && s.config.AWSSecretAccessKey != "" {
```

Comment:
I'm thinking if this condition should rather be:

```suggestion
    if s.config.AWSAccessKeyID != "" || s.config.AWSSecretAccessKey != "" || s.config.AWSSessionToken != "" {
```

Since all three values are used in that credentials provider.

Thinking out loud - does it ever make sense to provide a role _and_ the static credentials? Or should we disallow this to prevent a misconfiguration?

Reply:
The PR has been updated. The AWSAccessKeyID and AWSSecretAccessKey will be needed for static credentials. The IAM role config will be handled implicitly using AWS default authentication chain. See details in my PR description. 
```
// Only use static credentials if both access key and secret are provided
	if s.config.AWSAccessKeyID != "" && s.config.AWSSecretAccessKey != ""
```